client/src/pages/EditProfilePage/EditProfilePage.tsx

```diff
@@ -21,32 +22,167 @@ const EditProfilePage = () => {
+        firstName: profile.firstName || "",
+        linkedInProfile: profile.linkedInProfile || "",
+        gitHubProfile: profile.gitHubProfile || "",
+        nickName: profile.nickName || "",
         personalBio: profile.personalBio || "",
+        skills: profile.skills || [],
+        specializations: profile.specializations || [],
+        careerInformation: {
+          currentPosition: {
+            title: profile.careerInformation?.currentPosition?.title || "",
+            company: profile.careerInformation?.currentPosition?.company || "",
+          },
+        },
+        socialMediaLinks: {
+          twitter: profile.socialMediaLinks?.twitter || "",
+          blog: profile.socialMediaLinks?.blog || "",
+        },
+        availabilityForNetworking: profile.availabilityForNetworking || false,
       });
     }
   }, [profile]);
 
-  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const handleChange = (
+    e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
     const { name, value } = e.target;
     setFormData((prevFormData) => ({
       ...prevFormData,
       [name]: value,
     }));
   };
+  const handleCompanyChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      careerInformation: {
+        ...prevData.careerInformation,
+        currentPosition: {
+          ...prevData.careerInformation.currentPosition,
+          company: value,
+        },
+      },
+    }));
+  };
+  
+ const [skillInput, setSkillInput] = useState("");
+
+  const handleSkillChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setSkillInput(e.target.value);
+  };
+
+  const handleSpecialization = (skill: string) => {
+    setFormData((prevData) => {
+      const isAlreadySpecialized = prevData.specializations.includes(skill);
+      return isAlreadySpecialized
+        ? prevData
+        : {
+            ...prevData,
+            specializations: [...prevData.specializations, skill],
+          };
+    });
+  };
+
+  const handleSkillKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter" && skillInput.trim() !== "") {
+      e.preventDefault();
+      if (!formData.skills.includes(skillInput.trim())) {
+        setFormData((prevData) => ({
+          ...prevData,
+          skills: [...prevData.skills, skillInput.trim()],
+        }));
+      }
+      setSkillInput("");
+    }
+  };
+
+  const handleSkillRemove = (skill: string, specialty: boolean) => {
+    if (specialty) {
+      setFormData((prevData) => ({
+        ...prevData,
+        specializations: prevData.specializations.filter((s) => s !== skill),
+      }));
+    } else {
+      setFormData((prevData) => ({
+        ...prevData,
+        skills: prevData.skills.filter((s) => s !== skill),
+      }));
+    }
+  };
+
+  // Handling the change for title
+  const handleTitleChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
similar to consolidating twitter and blog changes below, handleTitleChange and handleCompanyChange both change state on the same nested object formData.careerInformation.currentPosition - could consolidate these 2 fn's into 1